app/src/main/java/org/oppia/app/player/state/ImageRegionSelectionInteractionView.kt

```diff
@@ -0,0 +1,91 @@
+package org.oppia.app.player.state
+
+import android.content.Context
+import android.os.Handler
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.databinding.BindingAdapter
+import org.oppia.app.model.ImageWithRegions
+import org.oppia.app.utility.ClickableAreasImage
+import org.oppia.app.utility.OnClickableAreaClickedListener
+import org.oppia.util.accessibility.CustomAccessibilityManager
+import org.oppia.util.gcsresource.DefaultResourceBucketName
+import org.oppia.util.parser.ExplorationHtmlParserEntityType
+import javax.inject.Inject
+
+class ImageRegionSelectionInteractionView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0
+) : AppCompatImageView(context, attrs, defStyleAttr) {
+
+  private var isAccessibilityEnabled: Boolean = false
+  private var clickableAreas: List<ImageWithRegions.LabeledRegion> = emptyList()
+
+  @Inject
+  lateinit var accessibilityManager: CustomAccessibilityManager
+
+  @Inject
+  @field:ExplorationHtmlParserEntityType
+  lateinit var entityType: String
+
+  @Inject
+  @field:DefaultResourceBucketName
+  lateinit var resourceBucketName: String
+
+  private lateinit var entityId: String
+
+  fun setEntityId(entityId: String) {
+    this.entityId = entityId
+  }
+
+  fun setClickableAreas(clickableAreas: List<ImageWithRegions.LabeledRegion>) {
+    this.clickableAreas = clickableAreas
+  }
+
+  fun getClickableAreas(): List<ImageWithRegions.LabeledRegion> {
+    return clickableAreas
+  }
+
+  fun isAccessibilityEnabled(): Boolean{
+    return isAccessibilityEnabled
+  }
+//
+//  override fun onAttachedToWindow() {
```

Comment:
Please don't commit commented out code. You can always commit code you plan to use, then commit removing it. Version history gives you the same benefit of commented out code. However, commented out code can quickly become out of date and lose context over time.

Reply:
actually this is something i needed to ask you about with this code uncommented i can't run this on a activity would i be needing a fragment to achieve this ?

Reply:
Is now addressed per our most recent meeting?

Reply:
Yup will be using by introducing a fragment